src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,185 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        private Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode;
+        if (isEmpty()) {
+            newNode = new Node<>(null, value, null);
+            head = tail = newNode;
+        } else {
+            newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Wrong index");
+        }
+        Node<T> newNode;
+        if (isEmpty()) {
+            newNode = new Node<>(null, value, null);
+            head = tail = newNode;
+            size++;
+            return;
+        }
+        if (index == 0) {
+            newNode = new Node<>(null, value, head);
+            head.prev = newNode;
+            head = newNode;
+            size++;
+            return;
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> currentNode = findNodeByIndex(index);
+        newNode = new Node<>(currentNode.prev, value, currentNode);
+        currentNode.prev.next = newNode;
+        currentNode.prev = newNode;
```

Comment:
Давай максимально спростимо логіку роботи методів. Те ж стосується і інших методів
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        Node<T> currentNode = findByIndex(index);
        Node<T> newNode = new Node<T>(currentNode.previous, value, currentNode);
        if (index == 0) {
            head = newNode;
        } else {
            currentNode.previous.next = newNode;
        }
        currentNode.previous = newNode;
```